SIL.Core.Tests/Extensions/StringExtensionTests.cs

```diff
@@ -431,5 +431,211 @@ public void SanitizePath_StringHasTwoTrailingDotsFollowingDirectorySeparator_NoC
+				"\u00BFEntonces que\u0301 " + StringExtensions.kObjReplacementChar + "?",
+				"\u00BFEntonces que\u0301 " + StringExtensions.kObjReplacementChar + "?", out fWholeWord, .15));
+			Assert.IsTrue(fWholeWord);
+		}
+
+		/// ------------------------------------------------------------------------------------
+		/// <summary>
+		/// Test the LongestUsefulCommonSubstring method. This test ensures that ORCs don't match
+		/// when matching partial words.
+		/// </summary>
+		/// ------------------------------------------------------------------------------------
+		[Test]
+		public void LongestUsefulCommonSubstring_PreventOrcMatching_PartialWordMatch()
+		{
+			bool fWholeWord;
+
+			Assert.AreEqual(" f", StringExtensions.GetLongestUsefulCommonSubstring(
+				"floppy " + StringExtensions.kObjReplacementChar + " friends",
+				"best " + StringExtensions.kObjReplacementChar + " forks", out fWholeWord, .15));
+			Assert.IsFalse(fWholeWord);
+        }
+
+        /// ------------------------------------------------------------------------------------
+        /// <summary>
+        /// Test the LongestUsefulCommonSubstring method. This test ensures that ORCs don't match
+        /// when matching partial words.
+        /// </summary>
+        /// ------------------------------------------------------------------------------------
+        [Test]
+        public void LongestUsefulCommonSubstring_PreventOrcMatching_OnlyOrcsAndWhitespaceMatch()
+        {
+            bool fWholeWord;
+
+            Assert.AreEqual(string.Empty, StringExtensions.GetLongestUsefulCommonSubstring(
+                StringExtensions.kObjReplacementChar + " " + StringExtensions.kObjReplacementChar + "ab?",
+                StringExtensions.kObjReplacementChar + " " + StringExtensions.kObjReplacementChar + "?", out fWholeWord, .15));
+            Assert.IsFalse(fWholeWord);
+        }
+
+        /// ------------------------------------------------------------------------------------
+        /// <summary>
+        /// Test the LongestUsefulCommonSubstring method. This test ensures that ORCs don't match
+        /// when matching partial words.
+        /// </summary>
+        /// ------------------------------------------------------------------------------------
+        [Test]
+        public void LongestUsefulCommonSubstring_PreventOrcMatching_OneStringIsAllOrcsAndSpaces()
+        {
+            bool fWholeWord;
+
+            Assert.AreEqual(string.Empty, StringExtensions.GetLongestUsefulCommonSubstring(
+                StringExtensions.kObjReplacementChar + " " + StringExtensions.kObjReplacementChar + " " + StringExtensions.kObjReplacementChar + " " + StringExtensions.kObjReplacementChar + " ",
+                StringExtensions.kObjReplacementChar + " " + StringExtensions.kObjReplacementChar + " " + StringExtensions.kObjReplacementChar + "a " + StringExtensions.kObjReplacementChar + " ", out fWholeWord, .15));
+            Assert.IsFalse(fWholeWord);
+        }
 	}
```

Comment:
I think it would be good to show the use as extension methods in one of the tests, and to add an additional test that tests with different values for `minPctForPartialWordMatch`.

Also, what happens if `minPctForPartialWordMatch` > 1 or `minPctForPartialWordMatch` < 0?